api/src/pcapi/core/subscription/api.py

```diff
@@ -412,3 +412,43 @@ def start_workflow(
     pcapi_repository.repository.save(user)
     fraud_check = fraud_api.start_fraud_check(user, thirdparty_id, content)
     return fraud_check
+
+
+# TODO (Lixxday): use a proper BeneficiaryFraudCHeck History model to track these kind of updates
+def handle_eligibility_difference_between_declaration_and_identity_provider(
+    fraud_check: fraud_models.BeneficiaryFraudCheck,
+) -> fraud_models.BeneficiaryFraudCheck:
+    declared_eligibility = fraud_check.eligibilityType
+    id_provider_detected_eligibility = fraud_api.get_eligibility_type(fraud_check.source_data())
+
+    if declared_eligibility == id_provider_detected_eligibility or id_provider_detected_eligibility is None:
+        return fraud_check
+
+    # Update fraud check by creating a new one with the correct eligibility
+    new_fraud_check = fraud_models.BeneficiaryFraudCheck(
+        user=fraud_check.user,
+        type=fraud_check.type,
+        thirdPartyId=fraud_check.thirdPartyId,
+        resultContent=fraud_check.resultContent,
+        status=fraud_check.status,
+        reason=fraud_check.reason,
+        reasonCodes=fraud_check.reasonCodes,
+        eligibilityType=id_provider_detected_eligibility,
+    )
+
+    # Cancel the old fraud check
+    fraud_check.status = fraud_models.FraudCheckStatus.CANCELED
+    if fraud_check.reason is None or fraud_check.reason == "":
```

Comment:
en python tu peux faire directment `if not fraud_check.reason`
pour gagner de la place et ne pas dupliquer des strings tu peux aussi mettre une variable "Eligibility type changed by the identity provider"
```
reason = "Coucou si jamais je dois être modifié il ne suffit de le faire qu'à un seul endroit
fraud_check.reason = reason if not fraud_check.reason else f"{separator} {reason}"
```